File: library/src/main/java/com/ebr163/inspector/rule/TextLengthRule.java
package com.ebr163.inspector.rule;

/**
 * Created by Bakht
 * on 09.01.2018.
 */

public class TextLengthRule extends AbstractRule<String> {

    private final int length;
    private final TextLength textLength;

    public enum TextLength {
        MORE, LESS, EQUAL, MORE_OR_EQUAL, LESS_OR_EQUAL
    }

    public TextLengthRule(int length, TextLength textLength, String errorMessage) {
        super(errorMessage);
        this.length = length;
        this.textLength = textLength;
    }

    @Override
    public boolean verify(String value) {
        if (value == null) {
            return false;
        }

        if (textLength == TextLength.EQUAL) {
            return value.length() == length;
        } else if (textLength == TextLength.LESS) {
            return value.length() < length;
        } else if (textLength == TextLength.MORE) {
            return value.length() > length;
        } else if (textLength == TextLength.LESS_OR_EQUAL) {
            return value.length() <= length;
        } else {
            return value.length() >= length;
        }
    }
}

File: library/src/main/java/com/ebr163/inspector/rule/TextLengthRule.kt
package com.ebr163.inspector.rule

/**
 * Created by Bakht
 * on 09.01.2018.
 */

class TextLengthRule(
        private val length: Int,
        private val textLength: TextLength,
        errorMessage: String
) : AbstractRule<String>(errorMessage) {

    enum class TextLength {
        MORE, LESS, EQUAL, MORE_OR_EQUAL, LESS_OR_EQUAL
    }

    override fun verify(value: String?): Boolean {
        if (value == null) {
            return false
        }

        return when (textLength) {
            TextLength.EQUAL -> value.length == length
            TextLength.LESS -> value.length < length
            TextLength.MORE -> value.length > length
            TextLength.LESS_OR_EQUAL -> value.length <= length
            else -> value.length >= length
        }
    }
}
